diff --git a/gettext-tools/tests/lang-csharp b/gettext-tools/tests/lang-csharp
index 55879abfa..a43e2d50a 100755
--- a/gettext-tools/tests/lang-csharp
+++ b/gettext-tools/tests/lang-csharp
@@ -1,6 +1,6 @@
 #! /bin/sh
 . "${srcdir=.}/init.sh"; path_prepend_ . ../src
-
+set -x
 # Test of gettext facilities in the C# language.
 # Assumes an fr_FR locale is installed.
 # Assumes the following packages are installed: mono, mcs.
@@ -154,7 +154,7 @@ ${DIFF} fr.po fr.po.new || Exit 1
 
 : ${MSGFMT=msgfmt}
 GETTEXTCSHARPLIBDIR=../../../gettext-runtime/intl-csharp \
-${MSGFMT} --csharp -d . -r prog -l fr fr.po || Exit 1
+${MSGFMT} --csharp -v -d . -r prog -l fr fr.po || Exit 1
 
 # Test which of the fr_FR locales are installed.
 : ${LOCALE_FR=fr_FR}
diff --git a/gettext-tools/tests/msgunfmt-csharp-1 b/gettext-tools/tests/msgunfmt-csharp-1
index fbbd5cced..e43625de1 100755
--- a/gettext-tools/tests/msgunfmt-csharp-1
+++ b/gettext-tools/tests/msgunfmt-csharp-1
@@ -1,6 +1,6 @@
 #! /bin/sh
 . "${srcdir=.}/init.sh"; path_prepend_ . ../src
-
+set -x
 # Test of --csharp option.
 
 # Test whether we can compile and execute C# programs.
@@ -52,12 +52,12 @@ EOF
 
 : ${MSGFMT=msgfmt}
 GETTEXTCSHARPLIBDIR=../../../gettext-runtime/intl-csharp \
-${MSGFMT} --csharp -d mu-cs-1 -r prog -l fr mu-cs-1/fr.po || Exit 1
+${MSGFMT} --csharp -v -d mu-cs-1 -r prog -l fr mu-cs-1/fr.po || Exit 1
 
 : ${MSGUNFMT=msgunfmt}
 GETTEXTCSHARPEXEDIR=../../src \
 GETTEXTCSHARPLIBDIR=../../../gettext-runtime/intl-csharp \
-${MSGUNFMT} --csharp -d mu-cs-1 -r prog -l fr -o mu-cs-1/prog.out || Exit 1
+${MSGUNFMT} --csharp -v -d mu-cs-1 -r prog -l fr -o mu-cs-1/prog.out || Exit 1
 
 : ${MSGCAT=msgcat}
 ${MSGCAT} -s -o mu-cs-1/prog.sort mu-cs-1/prog.out || Exit 1
